libs/RobotKit/include/RCLogger.h

```diff
@@ -0,0 +1,72 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "LogKit.h"
+#include "boost/sml.hpp"
+
+namespace leka::system::robot::sm {
+
+struct logger {
+	auto trim_namespaces(const char *str) const
+	{
+		auto sstr = std::string {str};
+
+		static const auto namespaces = std::to_array<const std::string>({
+			std::string {"leka::system::robot::"},
+			std::string {"boost::sml::back::"},
+			std::string {"boost::ext::sml::v1_1_4::back::"},
+		});
+
+		for (const auto &nmspc: namespaces) {
+			auto pos = std::string::npos;
+			while ((pos = sstr.find(nmspc)) != std::string::npos) {
+				sstr.erase(pos, nmspc.length());
+			}
+		}
+
+		return sstr;
+	}
+
+	template <typename T>
+	auto name()
+	{
+		auto name = trim_namespaces(boost::sml::aux::get_type_name<T>());
+		return name;
+		// return boost::sml::aux::get_type_name<T>();
+	}
+
+	template <typename SM, typename TEvent>
+	void log_process_event([[maybe_unused]] const TEvent &event)
+	{
+		auto e = name<TEvent>();
+		log_info("%s", e.c_str());
```

Comment:
Why use `log_info` when previously used `log_debug`?

Reply:
because log_debug is only for debug, thus it must be removed before being merged.

log_info can be used in production to "know" what's happening. it's only for the important things and must be seldom used.

if/when we use log files, only info, errors and warnings (if added) will be saved